pandas/_libs/groupby.pyx

```diff
@@ -125,6 +134,8 @@ cdef float64_t median_linear(float64_t* a, int n) noexcept nogil:
 
         j = 0
         for i in range(n):
+            if is_datetimelike and a[i] == NPY_NAT:
+                continue
             if a[i] == a[i]:
                 tmp[j] = a[i]
                 j += 1
```

Comment:
In the case of datetimelike, I think we can avoid the check `a[i] == a[i]` here, is that right?

Reply:
Thanks for the review, looking at surrounding code looks like you are right about avoiding the check, group_var does treat both checks as exclusive:

```Python
      elif is_datetimelike:
          # With group_var, we cannot just use _treat_as_na bc
          #  datetimelike dtypes get cast to float64 instead of
          #  to int64.
          isna_entry = val == NPY_NAT
      else:
          isna_entry = _treat_as_na(val, is_datetimelike)
```